src/parseSfc.ts

```diff
@@ -26,25 +27,47 @@ export interface FileError extends Error {
   file?: string
 }
 
-export function tryParseCustomBlock(
-  content: string,
-  filePath: string,
-  blockName: string): any {
-  try {
-    return JSON5.parse(content)
+export function tryParseCustomBlock(block: CustomBlock, filePath: string): any {
+  const lang = block.lang ?? 'any'
+  let valid = true
+  let err: Error = new Error()
+  let result: any = {}
+  if (lang === 'json') {
+    try {
+      result = JSON.parse(block.content)
+    }
+    catch (e) {
+      err = e
+      valid = false
+    }
   }
-  catch (err) {
+  else if (lang === 'json5' || lang === 'any') {
     try {
-      return YAML.parse(content)
+      result = JSON5.parse(block.content)
     }
-    catch {
-      const wrapped: FileError = new Error(`Invalid JSON/JSON5/YAML format of <${blockName}> content in ${filePath}\n${err.message}`)
-
-      // Store file path to provide useful information to downstream tools
-      // like friendly-errors-webpack-plugin
-      wrapped.file = filePath
+    catch (e) {
+      err = e
+      valid = false
+    }
+  }
 
-      throw wrapped
+  else if (lang === 'yaml' || (lang === 'any' && !valid)) {
+    try {
+      result = YAML.parse(block.content)
+    }
+    catch (e) {
+      err = e
+      valid = false
     }
   }
+
+  if (valid) return result
```

Comment:
Don't need to check valid, if this.error() emit it will stop at that point, so just `return result`